.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17.x
+
+    - name: check dependencies
+      run: |
+        go mod tidy -v
```

Comment:
ideally this can be a `.sh` script which can be called through `Makefile`

Reply:
just copy and pasting things, but agree

Reply:
done